xonsh/events.py

```diff
@@ -0,0 +1,169 @@
+"""
+Events for xonsh.
+
+In all likelihood, you want builtins.__xonsh_events__
+
+The best way to "declare" an event is something like::
+
+    __xonsh_events__.on_spam.doc("Comes with eggs")
+"""
+import abc
```

Comment:
This is what is causing things to explode, because other parts of the code base `import collections.abc as abc`, which is probably the wrong thing to do. Alagamate then starts to treat them like the same import.